serde/src/de/mod.rs

```diff
@@ -678,49 +845,140 @@ pub trait MapVisitor {
     /// The error type that can be returned if some error occurs during deserialization.
     type Error: Error;
 
-    /// This returns a `Ok(Some((key, value)))` for the next (key-value) pair in the map, or
-    /// `Ok(None)` if there are no more remaining items.
+    /// This returns `Ok(Some(key))` for the next key in the map, or `Ok(None)`
+    /// if there are no more remaining entries.
+    ///
+    /// `Deserialize` implementations should typically use
+    /// `MapVisitor::visit_key` or `MapVisitor::visit` instead.
+    fn visit_key_seed<K>(&mut self, seed: K) -> Result<Option<K::Value>, Self::Error>
```

Comment:
Me too. But the updates in serde/src/de/value.rs show it is the same amount of code for implementors. I will be putting together a comprehensive guide for writing Deserializers over the next two weeks and I will explain this thoroughly.